lib/srv/desktop/rdp/rdpclient/src/lib.rs

```diff
@@ -775,11 +775,7 @@ impl TryFrom<BitmapEvent> for CGOBitmap {
 
         // e.decompress consumes e, so we need to call it separately, after populating the fields
         // above.
-        let mut data = if e.is_compress {
-            e.decompress()?
-        } else {
-            e.data
-        };
+        let mut data = e.decompress()?;
```

Comment:
Can you add some explanation to this?
I would expect the fix to be the other way around:
only decompress if `is_compress`

Even the example code has this logic:
https://docs.rs/rdp-rs/0.1.0/rdp/core/event/struct.BitmapEvent.html#method.decompress

So why the change?

Reply:
The example code is bad. If you look here, you'll see that `decompress` actually handles both cases (when the compression flag is set and unset)

https://github.com/gravitational/rdp-rs/blob/2aa86a1962160deb4de581411413723ef334a0c7/src/core/event.rs#L92-L102

Additionally, you'll see the actual example client does an unconditional decompress:

https://github.com/gravitational/rdp-rs/blob/7ac880d7efb7f05efef3c84476f7c24f4053e0ea/src/bin/mstsc-rs.rs#L86

Reply:
Thank you